sotodlib/site_pipeline/multilayer_preprocess_tod.py

```diff
@@ -281,76 +301,62 @@ def main(configs_init: str,
         logger.warning(f"No observations returned from query: {query}")
 
     # clean up lingering files from previous incomplete runs
+    policy_dir_init = os.path.dirname(configs_init['archive']['policy']['filename']) + '/temp/'
+    policy_dir_proc = os.path.dirname(configs_proc['archive']['policy']['filename']) + '/temp_proc/'
     for obs in obs_list:
         obs_id = obs['obs_id']
-        pp_util.save_group_and_cleanup(obs_id, configs_init, context_init,
-                                       subdir='temp', remove=overwrite)
-        pp_util.save_group_and_cleanup(obs_id, configs_proc, context_proc,
-                                       subdir='temp_proc', remove=overwrite)
+        if os.path.exists(policy_dir_init):
+            found = False
+            for f in os.listdir(policy_dir_init):
+                if obs_id in f:
+                    found = True
+                    break
+
+            if found:
+                error = pp_util.save_group_and_cleanup(obs_id, configs_init, context_init,
+                                                       subdir='temp', remove=overwrite)
+                if error is not None:
+                    f = open(errlog, 'a')
+                    f.write(f'\n{time.time()}, cleanup error\n{error[0]}\n{error[2]}\n')
+                    f.close()
```

Comment:
Can this whole block be a single function that takes in `policy_dir`, `obs_id`, `config`, `context`, `subdir`, `remove`, and `errlog` so we can just call that line twice here (and in other places where its needed i.e. preprocess_tod, demod mapmaker)